### ISSUE TITLE
How to use the pytorch optimizer CyclicLR ?

### ISSUE THREAD

**Issue by zzzfinal0** [open-mmlab/mmdetection#1226]
I'd like to use the pytorch scheduler CyclicLR in torch/optim/lr_scheduler.py.  I tried this mmd-config.py but failed:
```
#optimizer
optimizer = dict(type='SGD', lr=0.02, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))
# learning policy
lr_config = dict(
    policy='CyclicLR',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    base_lr = 0.001)
```
How to edit the config file? Or How to add the CyclicLR policy in mmcv/runner/hooks/lr_updater.py? （I want to avoid editing mmcv library files）

**Comment by zzzfinal0** [comment]
Should I modify build_optimizer ？ In:
```
    if paramwise_options is None:
        return obj_from_dict(optimizer_cfg, torch.optim,
                             dict(params=model.parameters()))
    else:
        assert isinstance(paramwise_options, dict)
        ##Here goes my codes.
```

**Comment by hellock** [comment]
You can implement a `CyclicLRUpdateHook` in https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/lr_updater.py.

**Comment by zzzfinal0** [comment]
Alright.
